File: PHP/CalculoSimplesTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("CalculoSimples.php");

class CalculoSimplesTest extends TestCase {

    public function testCalculoSimples() {
        $obj = new CalculoSimples();
        //Test 1
        $want = 15.50;
        $got = $obj->calcTotalToPay([12, 1, 5.30], [16, 2, 5.10]);
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 51.40;
        $got = $obj->calcTotalToPay([13, 2, 15.30], [161, 4, 5.20]);
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 3
        $want = 30.20;
        $got = $obj->calcTotalToPay([1, 1, 15.10], [2, 1, 15.10]);
        echo "\nTest 3: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }

}

File: GO/CalculoSimples_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestCalculoSimples(t *testing.T) {
	tables := []struct {
		pecas1 []float64
		pecas2 []float64
		want   float64
	}{
		{[]float64{12, 1, 5.30}, []float64{16, 2, 5.10}, 15.50},
		{[]float64{13, 2, 15.30}, []float64{161, 4, 5.20}, 51.40},
		{[]float64{1, 1, 15.10}, []float64{2, 1, 15.10}, 30.20},
	}

	for _, table := range tables {
		got := calcTotalToPay(table.pecas1, table.pecas2)

		if got != table.want {
			t.Errorf("Waiting for this %.2f but the return was this: %.2f", table.want, got)
			//t.Errorf("Waiting for this %2f to %2f but the return was this: %2f", table.want, table.pecas1[2], got)
		}
	}
}
